app/services/payments/stripe_payment/create_session.rb

```diff
@@ -5,65 +5,73 @@
 module Payments
   module StripePayment
     class CreateSession < ApplicationService
-      include CartHelper
       include Rails.application.routes.url_helpers
       include Dry::Monads[:result, :do]
       class << self
         include Dry::Matcher.for(:call, with: Dry::Matcher::ResultMatcher)
       end
 
-      def initialize(user:)
+      def initialize(user:, order:)
         super()
         @user = user
+        @order = order
       end
 
       def call
-        yield create_customer
-        yield create_line_items
-        create_session
+        customer = yield create_customer
+        line_items = yield create_line_items
+        session = yield create_session(customer, line_items)
+
+        yield update_order(session)
+        Success session
       end
 
       private
 
       def create_customer
-        @customer = @user.customers.find_by(processor: :stripe)
-        unless @customer.present?
+        customer = @user.customers.find_by(processor: :stripe)
+        unless customer.present?
           stripe_customer = Stripe::Customer.create({ email: @user.email, name: user_name })
-          @customer = @user.customers.create!(processor: :stripe, processor_id: stripe_customer.id)
+          customer = @user.customers.create!(processor: :stripe, processor_id: stripe_customer.id)
         end
 
-        Success @customer
+        Success customer
       end
 
       def create_line_items
-        @line_items = @user.line_items.includes(:product, :product_discount).map do |line_item|
+        line_items = @order.order_items.map do |order_item|
           {
-            quantity: line_item.quantity,
+            quantity: order_item.quantity,
             price_data: {
-              currency: line_item.product.currency,
-              unit_amount: get_final_product_price(line_item),
+              currency: order_item.currency,
+              unit_amount: order_item.price,
               product_data: {
-                name: line_item.product.name,
+                name: order_item.product.name,
               },
             },
           }
         end
-        @line_items ? Success(@line_items) : Failure('Your shopping cart is empty')
+        line_items ? Success(line_items) : Failure('Your shopping cart is empty')
       end
 
-      def create_session
+      def create_session(customer, line_items)
         Success Stripe::Checkout::Session.create(
           {
-            customer: @customer.processor_id,
-            success_url: success_checkout_url,
+            customer: customer.processor_id,
+            success_url: order_url(@order.number),
             cancel_url: cart_url,
-            line_items: @line_items,
+            line_items:,
```

Comment:
If that field is empty we still need to define it there?

Reply:
Rubocop improves that way. The field is not empty but gets the value from the method parameter with the same name.

Reply:
This can be confusing